libtiny_client/src/state.rs

```diff
@@ -207,21 +293,23 @@ impl StateInner {
                         self.usermask = Some(usermask);
 
                         // Initialize channel state
-                        match utils::find_idx(&self.chans, |(s, _)| s == chan) {
+                        match utils::find_idx(&self.chans, |c| &c.name == chan) {
                             None => {
-                                self.chans.push((chan.to_owned(), HashSet::new()));
+                                self.chans.push(Chan::new(chan.to_owned()));
```

Comment:
This initializes the Chan state with NotJoined instead of Joined, which is a bug probably.

Reply:
I didn't understand this part. I'm assuming that when it's not found / `None` then the channel is not actually joined. Possibly an unexpected state?

Reply:
I can think of at least two cases where this can happen:

- User sends a `JOIN ...` message in the server tab, instead of using `/join`.
- The server for whatever reason adds the user to a channel. I think this happened to me before, though I don't remember the details.

I think both branches of this `match` should initialize the `Chan` state as `Joined`.

Reply:
Done!